App/static/js/content-recommendation.js

```diff
@@ -0,0 +1,379 @@
+function recommendBooks() {
+    let element = document.getElementById("force_graph");
+
+    if (element != null) {
+        element.remove();
+    }
+
+    let query = document.getElementById("user_query");
+
+    if (query.value === '') {
+        alert("Please enter some text before querying!")
+    }
+    else {
+        const url = `http://localhost:5000/recommend/${query.value}`
+        fetch(url)
+            .then(response => response.json())
+            .then(json => {
+                json['query'] = {
+                    'score': 0,
+                    'title': query.value,
+                    'pandas_index': -1,
+                }
+                generate_force_graph(json);
+            })
+    }
+}
+
+function generate_force_graph(recommendations) {
+    let links = [];
+    let nodes = {};
+    let width = window.innerWidth, height = 800;
+
+    // compute the distinct nodes from the links.
+    for (let key in recommendations) {
```

Comment:
@ZebraAlgebra this is the JS implementation of the force graph. Note, it only expects a dict of recommendations. Do console.log to see my assumptions on this (namely, one entry has a key value of "query", which don't apply for your case)